ast/compile.go

```diff
@@ -1259,12 +1259,14 @@ func (c *Compiler) compile() {
 		if c.Failed() {
 			return
 		}
-		for _, s := range c.after[s.name] {
-			err := c.runStageAfter(s.MetricName, s.Stage)
-			if err != nil {
+		for _, a := range c.after[s.name] {
+			if err := c.runStageAfter(a.MetricName, a.Stage); err != nil {
 				c.err(err)
```

Comment:
I'll make it just return here instead. It'll be simpler and more in line with how query compiler works, I think.